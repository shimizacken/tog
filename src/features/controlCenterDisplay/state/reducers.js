const initialState = {
    drivingInfo: '',
    warningInfo: ''
};

const controlCenterDisplay = (state = initialState, action) => {
    return state;
};

export { controlCenterDisplay };