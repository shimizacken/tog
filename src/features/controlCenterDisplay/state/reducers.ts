import type { ControlCenterDisplayState } from "../../../state/store.types";

const initialState: ControlCenterDisplayState = {
  drivingInfo: "",
  warningInfo: "",
};

export const controlCenterDisplay = (state = initialState) => {
  return state;
};
