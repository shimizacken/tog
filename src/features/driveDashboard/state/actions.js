import { SET_DRIVE_BUTTON_STATUS, SET_THROTTLE_SPEED, SET_DRIVE_BUTTON_SWITCHING_STATUS } from "./constants";
import DrivingButtonStates from "../services/ui/enums/drivingButtonStates";

export const setDrivingButtonState = drivingButtonState => ({
  type: SET_DRIVE_BUTTON_STATUS,
  drivingButtonState
});

export const setDrivingButtonSwitchingState = () => ({
  type: SET_DRIVE_BUTTON_SWITCHING_STATUS,
  drivingButtonState: DrivingButtonStates.SWITCHING
});

export const setThrottleSpeed = throttleSpeed => ({
  type: SET_THROTTLE_SPEED,
  throttleSpeed,
  meta: {
    throttle: 500
  }
});
