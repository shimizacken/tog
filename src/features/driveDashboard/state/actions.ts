import {
  SET_DRIVE_BUTTON_STATUS,
  SET_THROTTLE_SPEED,
  SET_DRIVE_BUTTON_SWITCHING_STATUS,
  SET_DRIVING_STATUS,
} from "./constants";
import { DrivingButtonStates } from "../services/ui/enums/drivingButtonStates";

export interface SetDrivingButtonState {
  type: string;
  drivingButtonState: DrivingButtonStates;
}

export interface SetDrivingButtonSwitchingState {
  type: string;
  throttleSpeed: number;
  meta: {
    throttle: number;
  };
}

export interface SetDrivingStatus {
  type: string;
  drivingStatus: boolean;
}

export const setDrivingButtonState = (
  drivingButtonState: DrivingButtonStates
): SetDrivingButtonState => ({
  type: SET_DRIVE_BUTTON_STATUS,
  drivingButtonState,
});

export const setDrivingButtonSwitchingState = (): SetDrivingButtonState => ({
  type: SET_DRIVE_BUTTON_SWITCHING_STATUS,
  drivingButtonState: DrivingButtonStates.SWITCHING,
});

export const setThrottleSpeed = (
  throttleSpeed: number
): SetDrivingButtonSwitchingState => ({
  type: SET_THROTTLE_SPEED,
  throttleSpeed,
  meta: {
    throttle: 10,
  },
});

export const setDrivingStatus = (drivingStatus: boolean): SetDrivingStatus => ({
  type: SET_DRIVING_STATUS,
  drivingStatus,
});
