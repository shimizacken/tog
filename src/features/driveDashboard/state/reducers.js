const DrivingButtonStates = {
    TURNED_OFF: 0,
    READY: 1,
    DRIVE: 2,
    ERROR: 3
};

const initialState = {
    drivingButtonState: DrivingButtonStates.READY,
    throttleStick: 0
};

const driveDashboard = (state = initialState, action) => {
    return state;
 };

 export { driveDashboard };