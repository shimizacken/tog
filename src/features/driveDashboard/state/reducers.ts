import {
  SET_DRIVE_BUTTON_STATUS,
  SET_THROTTLE_SPEED,
  SET_DRIVE_BUTTON_SWITCHING_STATUS,
  SET_DRIVING_STATUS,
} from "./constants";
import { DrivingButtonStates } from "../services/ui/enums/drivingButtonStates";
import {
  SetDrivingButtonState,
  SetDrivingButtonSwitchingState,
  SetDrivingStatus,
} from "./actions";
import { Action } from "redux";
import type { DriveDashboardState } from "../../../state/store.types";

const driveDashboardInitialState: DriveDashboardState = {
  drivingButtonState: DrivingButtonStates.TURNED_OFF,
  throttleSpeed: 0,
  drivingStatus: false,
};

export const driveDashboard = (
  state = driveDashboardInitialState,
  action: Action
) => {
  if (action.type === SET_DRIVE_BUTTON_STATUS) {
    return {
      ...state,
      drivingButtonState: (action as SetDrivingButtonState).drivingButtonState,
    };
  }

  if (action.type === SET_DRIVE_BUTTON_SWITCHING_STATUS) {
    return {
      ...state,
      drivingButtonState: (action as SetDrivingButtonState).drivingButtonState,
    };
  }

  if (action.type === SET_THROTTLE_SPEED) {
    return {
      ...state,
      throttleSpeed: (action as SetDrivingButtonSwitchingState).throttleSpeed,
    };
  }

  if (action.type === SET_DRIVING_STATUS) {
    return {
      ...state,
      drivingStatus: (action as SetDrivingStatus).drivingStatus,
    };
  }

  return state;
};
