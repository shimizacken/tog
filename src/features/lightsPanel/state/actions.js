import { SET_FRONT_LIGHT_STATE, SET_BACK_LIGHT_STATE, SET_CABIN_LIGHT_STATE, SET_WAGON_LIGHT_STATE } from "./constants";

export const setFrontLightState = frontLightsState => ({
  type: SET_FRONT_LIGHT_STATE,
  frontLightsState
});

export const setBackLightState = backLightsState => ({
    type: SET_BACK_LIGHT_STATE,
    backLightsState
});

export const setCabinLightState = cabinLights => ({
    type: SET_CABIN_LIGHT_STATE,
    cabinLights
});

export const setWagonsLightState = wagonsLights => ({
    type: SET_WAGON_LIGHT_STATE,
    wagonsLights
});
