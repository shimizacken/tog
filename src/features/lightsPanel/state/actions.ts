import {
  SET_FRONT_LIGHT_STATE,
  SET_BACK_LIGHT_STATE,
  SET_CABIN_LIGHT_STATE,
  SET_WAGON_LIGHT_STATE,
} from "./constants";

export const setFrontLightState = {
  type: SET_FRONT_LIGHT_STATE,
};

export const setBackLightState = {
  type: SET_BACK_LIGHT_STATE,
};

export const setCabinLightState = {
  type: SET_CABIN_LIGHT_STATE,
};

export const setWagonsLightState = {
  type: SET_WAGON_LIGHT_STATE,
};
