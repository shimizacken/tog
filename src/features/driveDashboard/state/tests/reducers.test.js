import { driveDashboard } from '../reducers';
import { setDrivingButtonState, setThrottleSpeed } from '../actions';
import DrivingButtonStates from '../../services/ui/drivingButtonStates';

describe('driveDashboard reducer test suite', () => {
  
    describe('initial state test suite', () => {
      
        it('should return initial state', () => {
            
            let state;
            const action = {
                type: '@@RANDOM_ACTION_TYPE'
            };
    
            expect(driveDashboard(state, action)).toMatchSnapshot();
        });
    });

    describe('SET_DRIVE_BUTTON_STATUS test suite', () => {
        
        it('should set DRIVE with initial state', () => {
            
            let state;
                    
            expect(driveDashboard(state, setDrivingButtonState(DrivingButtonStates.DRIVE))).toMatchSnapshot();
        });
    
        it('should set ERROR with initial state', () => {
            
            let state;
                    
            expect(driveDashboard(state, setDrivingButtonState(DrivingButtonStates.ERROR))).toMatchSnapshot();
        });
    
        it('should set TURNED_OFF with initial state', () => {
            
            let state;
                    
            expect(driveDashboard(state, setDrivingButtonState(DrivingButtonStates.TURNED_OFF))).toMatchSnapshot();
        });
    
        it('should set READY with none initial state', () => {
            
            const state = {
                drivingButtonState: DrivingButtonStates.DRIVE,
                throttleSpeed: 0
            };
                    
            expect(driveDashboard(state, setDrivingButtonState(DrivingButtonStates.READY))).toMatchSnapshot();
        });
    });
    
    describe('SET_THROTTLE_SPEED test suite', () => {
      
        it('should set throttle speed to 20 with initial state', () => {
            
            let state;

            expect(driveDashboard(state, setThrottleSpeed(20))).toMatchSnapshot();
        });

        it('should set throttle speed to 100 with none initial state', () => {
            
            const state = {
                drivingButtonState: DrivingButtonStates.DRIVE,
                throttleSpeed: 50
            };

            expect(driveDashboard(state, setThrottleSpeed(100))).toMatchSnapshot();
        });
    });
});
