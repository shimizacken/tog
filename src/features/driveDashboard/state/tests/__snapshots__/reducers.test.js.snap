// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`driveDashboard reducer test suite SET_DRIVE_BUTTON_STATUS test suite should set DRIVE with initial state 1`] = `
Object {
  "drivingButtonState": "DRIVE",
  "throttleSpeed": 0,
}
`;

exports[`driveDashboard reducer test suite SET_DRIVE_BUTTON_STATUS test suite should set ERROR with initial state 1`] = `
Object {
  "drivingButtonState": "ERROR",
  "throttleSpeed": 0,
}
`;

exports[`driveDashboard reducer test suite SET_DRIVE_BUTTON_STATUS test suite should set READY with none initial state 1`] = `
Object {
  "drivingButtonState": "READY",
  "throttleSpeed": 0,
}
`;

exports[`driveDashboard reducer test suite SET_DRIVE_BUTTON_STATUS test suite should set TURNED_OFF with initial state 1`] = `
Object {
  "drivingButtonState": "TURNED_OFF",
  "throttleSpeed": 0,
}
`;

exports[`driveDashboard reducer test suite SET_THROTTLE_SPEED test suite should set throttle speed to 20 with initial state 1`] = `
Object {
  "drivingButtonState": "TURNED_OFF",
  "throttleSpeed": 20,
}
`;

exports[`driveDashboard reducer test suite SET_THROTTLE_SPEED test suite should set throttle speed to 100 with none initial state 1`] = `
Object {
  "drivingButtonState": "DRIVE",
  "throttleSpeed": 100,
}
`;

exports[`driveDashboard reducer test suite initial state test suite should return initial state 1`] = `
Object {
  "drivingButtonState": "TURNED_OFF",
  "throttleSpeed": 0,
}
`;
